vyxal/elements.py

```diff
@@ -248,6 +270,31 @@ def all_diagonals(lhs, ctx):
     return all_diags
 
 
+def all_diagonals_ordered(lhs, ctx):
```

Comment:
Do we want to add support for LazyLists? We can't make this work for an infinite list of infinite lists, but I think it could be modified to work for an infinite list of finite lists.